core/templates/services/playthrough.service.ts

```diff
@@ -249,53 +270,63 @@ export class PlaythroughService {
         state_name: {value: stateName},
         time_spent_in_state_in_msecs: {value: 1000 * timeSpentInStateSecs}
       }));
-
-    this.playthroughDurationInSecs = this.playthroughStopwatch.getTimeInSecs();
-    this.eqTracker.recordExplorationQuit(
-      stateName, this.playthroughDurationInSecs);
+    if (this.playthroughStopwatch !== null && this.eqTracker !== null) {
+      this.playthroughDurationInSecs = (
+        this.playthroughStopwatch.getTimeInSecs());
+      this.eqTracker.recordExplorationQuit(
+        stateName, this.playthroughDurationInSecs);
+    }
   }
 
   storePlaythrough(): void {
     if (this.isRecordedPlaythroughHelpful()) {
       const playthrough = this.createNewPlaythrough();
       if (playthrough !== null) {
-        this.playthroughBackendApiService.storePlaythroughAsync(playthrough, 1);
+        this.playthroughBackendApiService.storePlaythroughAsync(
+          playthrough, 1);
       }
     }
   }
 
   /**
-   * The ordering of checks in this method prioritizes the following types of
-   * playthroughs:
-   *    1. MultipleIncorrectSubmissionsIssue
-   *    2. CyclicStateTransitionsIssue
-   *    3. EarlyQuitIssue
-   *
-   * If none of the issue types have been discovered, returns null instead.
-   */
+  * The ordering of checks in this method prioritizes the following types of
+  * playthroughs:
+  *    1. MultipleIncorrectSubmissionsIssue
+  *    2. CyclicStateTransitionsIssue
+  *    3. EarlyQuitIssue
+  *
+  * If none of the issue types have been discovered, returns null instead.
+  */
   private createNewPlaythrough(): Playthrough | null {
-    if (this.misTracker && this.misTracker.foundAnIssue()) {
-      return this.playthroughObjectFactory
-        .createNewMultipleIncorrectSubmissionsPlaythrough(
-          this.explorationId, this.explorationVersion,
-          this.misTracker.generateIssueCustomizationArgs(),
-          this.recordedLearnerActions);
-    } else if (this.cstTracker && this.cstTracker.foundAnIssue()) {
-      return this.playthroughObjectFactory
-        .createNewCyclicStateTransitionsPlaythrough(
-          this.explorationId, this.explorationVersion,
-          this.cstTracker.generateIssueCustomizationArgs(),
-          this.recordedLearnerActions);
-    } else if (this.eqTracker && this.eqTracker.foundAnIssue()) {
-      return this.playthroughObjectFactory
-        .createNewEarlyQuitPlaythrough(
-          this.explorationId, this.explorationVersion,
-          this.eqTracker.generateIssueCustomizationArgs(),
-          this.recordedLearnerActions);
+    if (
```

Comment:
Why is it possible for these quantities be null?

It seems like it should be an error if this function is called when they are null... I suspect you might need to do a bit more analysis.

Reply:
I had opened previous Prs and there it does not seem to be a problem there